Tiltfile

```diff
@@ -411,74 +411,89 @@ def prepare_all():
+        label = p.get("label", provider)
+
         for template_dir in provider_dirs:
             template_list = [filename for filename in listdir(template_dir) if os.path.basename(filename).endswith("yaml")]
             for filename in template_list:
-                deploy_templates(filename, provider, substitutions)
+                deploy_templates(filename, label, substitutions)
 
-def deploy_templates(filename, provider, substitutions):
+def deploy_templates(filename, label, substitutions):
     # validate filename exists
     if not os.path.exists(filename):
         fail(filename + " not found")
 
-    os.environ["NAMESPACE"] = substitutions.get("NAMESPACE", "default")
-    os.environ["KUBERNETES_VERSION"] = substitutions.get("KUBERNETES_VERSION", "v1.24.0")
-    os.environ["CONTROL_PLANE_MACHINE_COUNT"] = substitutions.get("CONTROL_PLANE_MACHINE_COUNT", "1")
-    os.environ["WORKER_MACHINE_COUNT"] = substitutions.get("WORKER_MACHINE_COUNT", "3")
-
     basename = os.path.basename(filename)
     if basename.endswith(".yaml"):
         if basename.startswith("clusterclass-"):
             template_name = basename.replace("clusterclass-", "").replace(".yaml", "")
-            deploy_clusterclass(template_name, provider, filename)
+            deploy_clusterclass(template_name, label, filename, substitutions)
         elif basename.startswith("cluster-template-"):
             clusterclass_name = basename.replace("cluster-template-", "").replace(".yaml", "")
-            deploy_cluster_template(clusterclass_name, provider, filename)
+            deploy_cluster_template(clusterclass_name, label, filename, substitutions)
 
-def deploy_clusterclass(clusterclass_name, provider, filename):
+def deploy_clusterclass(clusterclass_name, label, filename, substitutions):
     apply_clusterclass_cmd = "cat " + filename + " | " + envsubst_cmd + " | " + kubectl_cmd + " apply -f - && echo \"ClusterClass created from\'" + filename + "\', don't forget to delete\n\""
-    delete_clusterclass_cmd = kubectl_cmd + " delete clusterclass " + clusterclass_name + ' --ignore-not-found=true; echo "\n"'
+    delete_clusterclass_cmd = kubectl_cmd + " --namespace=$NAMESPACE delete clusterclass " + clusterclass_name + ' --ignore-not-found=true; echo "\n"'
 
     local_resource(
         name = clusterclass_name,
         cmd = ["bash", "-c", apply_clusterclass_cmd],
+        env = substitutions,
         auto_init = False,
         trigger_mode = TRIGGER_MODE_MANUAL,
-        labels = [provider + "-clusterclasses"],
+        labels = [label + ".clusterclasses"],
     )
 
     cmd_button(
         clusterclass_name + ":apply",
         argv = ["bash", "-c", apply_clusterclass_cmd],
         resource = clusterclass_name,
         icon_name = "note_add",
-        text = "Apply ClusterClass",
+        text = "Apply `" + clusterclass_name + "` ClusterClass",
+        inputs = [
+            text_input("NAMESPACE", default = substitutions.get("NAMESPACE")),
```

Comment:
Looks like it is not possible to deploy a ClusterClass to a different namespace by setting the `Namespace` value here. 
This is because the clusterclass template does not have any variables which also means it does not support setting namespace. It is recommended that clusterclass definitions shoud not inlucde namespace (ref: https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#target-namespace-2) to enable support with `clusterctl generate cluster`.

With this shoud we drop this field?

This would also mean a user cannot deploy a Cluster to a different namespace using the corresponding buttons unless the user manually deploys the ClusterClass to that namespace.

Reply:
I think instead we should at `--namespace=$NAMESPACE` to the `apply_clusterclass_cmd` as done for the `delete_clusterclass_cmd` :-)

Reply:
(still requires this input field then)

Reply:
Good idea. That would work with ClusterClass and different namespaces and we can keep the field.

Reply:
done